ambari-server/src/main/resources/stack-hooks/before-ANY/scripts/params.py

```diff
@@ -146,59 +151,59 @@ def is_secure_port(port):
     hadoop_secure_dn_user = '""'
 
 #hadoop params
-hdfs_log_dir_prefix = config['configurations']['hadoop-env']['hdfs_log_dir_prefix']
-hadoop_pid_dir_prefix = config['configurations']['hadoop-env']['hadoop_pid_dir_prefix']
-hadoop_root_logger = config['configurations']['hadoop-env']['hadoop_root_logger']
+hdfs_log_dir_prefix = module_configs.get_property_value(module_name, 'hadoop-env', 'hdfs_log_dir_prefix')
+hadoop_pid_dir_prefix = module_configs.get_property_value(module_name, 'hadoop-env', 'hadoop_pid_dir_prefix')
+hadoop_root_logger = module_configs.get_property_value(module_name, 'hadoop-env', 'hadoop_root_logger')
 
 jsvc_path = "/usr/lib/bigtop-utils"
 
-hadoop_heapsize = config['configurations']['hadoop-env']['hadoop_heapsize']
-namenode_heapsize = config['configurations']['hadoop-env']['namenode_heapsize']
-namenode_opt_newsize = config['configurations']['hadoop-env']['namenode_opt_newsize']
-namenode_opt_maxnewsize = config['configurations']['hadoop-env']['namenode_opt_maxnewsize']
-namenode_opt_permsize = format_jvm_option("/configurations/hadoop-env/namenode_opt_permsize","128m")
-namenode_opt_maxpermsize = format_jvm_option("/configurations/hadoop-env/namenode_opt_maxpermsize","256m")
+hadoop_heapsize = module_configs.get_property_value(module_name, 'hadoop-env', 'hadoop_heapsize')
```

Comment:
Similar case, what happens if module_name is zookeeper?

Reply:
This would be addressed based on service group dependencies and service instance dependencies 

For example:
ODS1/HBASE -> ODS1/HADOOP_CLIENT -> HDPCORE1/HDFS 
ODS2/HBASE -> ODS2/HADOOP_CLIENT -> HDPCORE2/HDFS

We will have to revisit this once we refactor our execution command to deal with multiple instances but for now we should can ignore the module_name since there is only one hadoop-env setting in the cluster.